openstack/clustering/v1/policies/requests.go

```diff
@@ -54,3 +54,46 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 		return p
 	})
 }
+
+// CreateOpts params
+type CreateOpts struct {
+	Name string `json:"name"`
+	Spec Spec   `json:"-"`
+}
+
+type Spec struct {
+	Description string
+	Properties  map[string]interface{}
+	Type        string
+	Version     string
+}
+
+// ToPolicyCreateMap formats a CreateOpts into a body map.
+func (opts CreateOpts) ToPolicyCreateMap() (map[string]interface{}, error) {
+	b, err := gophercloud.BuildRequestBody(opts, "")
+	if err != nil {
+		return nil, err
+	}
+
+	spec := make(map[string]interface{})
+	spec["description"] = opts.Spec.Description
+	spec["type"] = opts.Spec.Type
+	spec["version"] = opts.Spec.Version
+	spec["properties"] = opts.Spec.Properties
+	b["spec"] = spec
```

Comment:
3. Comment this entire block out.

With 1, 2, and 3 in place, everything marshals correctly for me. Are there occurrences when that won't happen and the above method is needed?

Reply:
I changed it and It works correctly.